modules/openideal_user/src/Form/OpenidealSecondStepRegistrationForm.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\openideal_user\Form;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Url;
+
+/**
+ * OpenidealSecondStepRegistration object.
+ */
+class OpenidealSecondStepRegistrationForm extends ContentEntityForm {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'user-additional-details';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $user = $this->entityTypeManager->getStorage('user')->load($this->currentUser()->id());
+    self::setEntity($user);
+
+    /* @var $entity \Drupal\user\Entity\User */
+    $form = parent::buildForm($form, $form_state);
+
+    $last_name = $user->get('field_last_name')->getString();
+    $first_name = $user->get('field_first_name')->getString();
+    if (!empty($first_name)) {
+      $form['field_last_name']['widget']['0']['value']['#required'] = FALSE;
+      $form['field_last_name']['#type'] = 'hidden';
```

Comment:
If you'd like to hide the field, then it's better to use `#access` property:
https://api.drupal.org/api/drupal/developer%21topics%21forms_api_reference.html/7.x#access